x-pack/plugins/fleet/server/routes/package_policy/handlers.ts

```diff
@@ -242,19 +243,24 @@ export const createPackagePolicyHandler: FleetRequestHandler<
   const authorizationHeader = HTTPAuthorizationHeader.parseFromRequest(request, user?.username);
   let wasPackageAlreadyInstalled = false;
 
-  if ('output_id' in newPolicy) {
-    // TODO Remove deprecated APIs https://github.com/elastic/kibana/issues/121485
-    delete newPolicy.output_id;
-  }
   const spaceId = fleetContext.spaceId;
   try {
     if (!newPolicy.policy_id && (!newPolicy.policy_ids || newPolicy.policy_ids.length === 0)) {
       throw new PackagePolicyRequestError('Either policy_id or policy_ids must be provided');
     }
 
-    const { canUseReusablePolicies, errorMessage } = canUseMultipleAgentPolicies();
+    const { canUseReusablePolicies, errorMessage: canUseMultipleAgentPoliciesErrorMessage } =
+      canUseMultipleAgentPolicies();
     if ((newPolicy.policy_ids ?? []).length > 1 && !canUseReusablePolicies) {
-      throw new PackagePolicyRequestError(errorMessage);
+      throw new PackagePolicyRequestError(canUseMultipleAgentPoliciesErrorMessage);
+    }
+
+    if (newPolicy.output_id && pkg) {
```

Comment:
Should that validation happen at the package policy service level so it will also be validated for preconfigured package policies and in the future if other solution create package policy with `output_id` 

Should we validate also for the update package policy method? (Maybe it's already done and I missed that)

Reply:
yes, this should happen at the service level. I was copying what was done for multiple policy IDs but I don't think that was a good pattern. I pushed changes to move both multiple policy and integration output validation to the service level. they are executed together in `preflight` function that is added to create, update, bulk create, and bulk update.

thanks for catching this pattern!